finances/views.py

```diff
@@ -1843,30 +1839,41 @@ def form_valid(self, form, *args, **kwargs):
             isParticipant = False
 
         errors = []
+        nb_empty_rows = 0
         i = 1
 
         # Enregistrement des pondérations
         for row in rows:
             i += 1;
             try:
-                username = row[0].value.strip() # Should be a str
-                user = User.objects.get(username=username)
-                try:
-                    pond = int(row[1].value) # Should be an int. Else, raise an error
-
-                    if pond > 0:
-                        self.se.change_weight( user, pond, isParticipant )
-                except:
-                    errors.append( "Erreur avec " + username + " (ligne n*" + str(i) + "). A priori pas ajouté." )
+                if row[0].value and row[1].value:
+                  username = row[0].value.strip() # Should be a str
+                  user = User.objects.filter(username=username)
+                  if user.count() > 0:
+                      user = User.objects.get(username=username)
+                      try:
+                          pond = int(row[1].value) # Should be an int. Else, raise an error
+                          if pond > 0:
+                              self.se.change_weight( user, pond, isParticipant )
+                      except:
+                          errors.append( "Erreur avec " + username + " (ligne n*" + str(i) + "). A priori pas ajouté." )
+                  else:
+                      errors.append( "L'utilisateur " + username + " n'existe pas. (ligne n*" + str(i) + ").")
+                else:
+                  nb_empty_rows += 1
```

Comment:
If you don't add an error there, you are losing the line data

Reply:
On which line ?

Reply:
I just realized a manager can import a excel with all the users, and only add some weight for only a few of them. So my bad, it's fine like this